File: library/src/main/java/com/afollestad/aesthetic/BgIconColorState.java
package com.afollestad.aesthetic;

import android.support.annotation.ColorInt;
import android.support.annotation.Nullable;
import android.support.annotation.RestrictTo;

import rx.functions.Func2;

import static android.support.annotation.RestrictTo.Scope.LIBRARY_GROUP;

/** @author Aidan Follestad (afollestad) */
@RestrictTo(LIBRARY_GROUP)
final class BgIconColorState {

  private final int bgColor;
  private final ActiveInactiveColors iconTitleColor;

  private BgIconColorState(int bgColor, ActiveInactiveColors iconTitleColor) {
    this.bgColor = bgColor;
    this.iconTitleColor = iconTitleColor;
  }

  static BgIconColorState create(int color, ActiveInactiveColors iconTitleColors) {
    return new BgIconColorState(color, iconTitleColors);
  }

  @ColorInt
  int bgColor() {
    return bgColor;
  }

  @Nullable
  ActiveInactiveColors iconTitleColor() {
    return iconTitleColor;
  }

  static Func2<Integer, ActiveInactiveColors, BgIconColorState> creator() {
    return new Func2<Integer, ActiveInactiveColors, BgIconColorState>() {
      @Override
      public BgIconColorState call(Integer integer, ActiveInactiveColors activeInactiveColors) {
        return BgIconColorState.create(integer, activeInactiveColors);
      }
    };
  }
}

File: library/src/main/java/com/afollestad/aesthetic/BgIconColorState.kt
package com.afollestad.aesthetic

import android.support.annotation.ColorInt
import io.reactivex.functions.BiFunction

/** @author Aidan Follestad (afollestad) */
internal data class BgIconColorState(
  @field:ColorInt val bgColor: Int,
  val iconTitleColor: ActiveInactiveColors?
) {

  companion object {
    fun creator(): BiFunction<Int, ActiveInactiveColors, BgIconColorState> {
      return BiFunction { color, iconTitleColors -> BgIconColorState(color, iconTitleColors) }
    }
  }
}
